app/services/master_controller.py

```
@@ -2388,6 +2409,220 @@ async def set_trading_mode(self, mode: str) -> Dict[str, Any]:
+                weights=ai_model_weights,
+                frequency=autonomous_frequency_minutes,
+                base_mode=current_mode.value
+            )
+            
+            return {
+                "success": True,
+                "message": "AI model weights updated successfully",
+                "ai_model_weights": ai_model_weights,
+                "autonomous_frequency_minutes": autonomous_frequency_minutes or base_config.autonomous_frequency_minutes,
+                "base_trading_mode": current_mode.value
+            }
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)